salt/utils/http.py

```diff
@@ -460,6 +461,13 @@ def query(url,
         proxy_port = opts.get('proxy_port', None)
         proxy_username = opts.get('proxy_username', None)
         proxy_password = opts.get('proxy_password', None)
+        no_proxy = opts.get('proxy_host', [])
+
+        # Since tornado doesnt support no_proxy, we'll always hand it empty proxies or valid ones
+        # except we remove the valid ones if a url has a no_proxy hostname in it
+        if urlparse(url_full).hostname in no_proxy:
```

Comment:
I think this will not work with subdomain globs (e.g. *.cluster.local), which usually works afaik in `no_proxy`.